### PR TITLE
Add grammars to CFPQ_Data

### PR DESCRIPTION
**Pull request by rustam-azimov**

### CODE REVIEW

**Comment by rustam-azimov** [comment]
**Main contribution for review:**
- whole website (except `Benchmarks` that is demo for now) and especially `Grammars` and `Tutorial` pages,
- `download_grammars` function in `dataset/data` module,
- whole `grammars/generators` module for 4 new grammar templates.

For generating and open the website please use the following command (instead of `chromium ` you should use your browser command):
```
 make -C docs/ clean && make -C docs/ html && chromium docs/_build/html/index.html
```

File: cfpq_data/grammars/generators/nested_parentheses_grammar.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    start_symbol : Variable
+        Start symbol of the grammar.
+
+    Examples
+    --------
+    >>> from cfpq_data import *
+    >>> path = download("foaf")
+    >>> g = cfpq_data.graph_from_csv(path)
+    >>> cfg = nested_parentheses_grammar_from_graph(g, n_types=2)
```
[the remark this answers is not in the document]
Comment (gsvgit) [reply cfpq_data/grammars/generators/nested_parentheses_grammar.py:101]: Moreover, what if I want to use relation `a`as an open parenthesis and relation `b` as a close one?

**Review by gsvgit — COMMENTED** [review]
Generation of Dyck-like languages should be described carefully.

File: cfpq_data/graphs/utils/edges_statistics.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    >>> labels_frequency = get_labels_frequency(g)
+    >>> labels_frequency
+    {'a': 2, 'b': 2}
+
+    Returns
+    -------
+    labels_frequency : Dict[Any, int]
+        Dictionary with edge labels usage frequency.
+    """
+    labels_frequency = dict()
```
Comment (vadyushkins) [inline cfpq_data/graphs/utils/edges_statistics.py:33]: I think `defaultdict` would be better.
```suggestion
    labels_frequency = defaultdict(int)
```

File: cfpq_data/graphs/utils/edges_statistics.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    logging.info(f"Construct {labels_frequency=} for {graph=}")
+
+    return labels_frequency
+
+
+def get_sorted_labels(
+    graph: nx.MultiDiGraph,
+    *,
+    ascending: bool = False,
```
Comment (vadyushkins) [inline cfpq_data/graphs/utils/edges_statistics.py:47]: I think `reverse` would be better.

**Review by vadyushkins — CHANGES_REQUESTED** [review]
I think we don't need functions like `nested_parentheses_grammar_from_graph`. The naive function `nested_parentheses_grammar` function is enough for the user. Same for `dyck_grammar_from_graph` and `java_points_to_grammar_from_graph`.

File: cfpq_data/grammars/generators/nested_parentheses_grammar.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    >>> cfg_to_text(cfg)
+    'S -> \\nS -> label S label_r\\nS -> type S type_r'
```
[the remark this answers is not in the document]
Comment (rustam-azimov) [reply cfpq_data/grammars/generators/nested_parentheses_grammar.py:103]: S -> \\n here is for S -> epsilon

File: cfpq_data/grammars/generators/nested_parentheses_grammar.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rustam-azimov) [reply cfpq_data/grammars/generators/nested_parentheses_grammar.py:101]: Yes, I think we don't need this function.

File: cfpq_data/graphs/utils/edges_statistics.py (hunk 2 above)
Comment (rustam-azimov) [reply cfpq_data/graphs/utils/edges_statistics.py:33]: Yes, I agree for `nested_parentheses_grammar_from_graph` and `dyck_grammar_from_graph`. However, `java_points_to_grammar_from_graph` automatically finds set of fields used in java programs corresponding to our java graphs. For now, only web description of java graphs have information about which fields they have. You think `java_points_to_grammar_from_graph` should be also removed?

File: cfpq_data/graphs/utils/edges_statistics.py (hunk 3 above)
Comment (rustam-azimov) [reply cfpq_data/graphs/utils/edges_statistics.py:47]: Fixed

File: cfpq_data/graphs/utils/edges_statistics.py (hunk 2 above)
Comment (rustam-azimov) [reply cfpq_data/graphs/utils/edges_statistics.py:33]: defaultdict fixed.

File: .github/ISSUE_TEMPLATE/grammar-add-template.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+---
+name: ADD NEW GRAMMAR
+about: Suggest new grammar template to this project
+title: "[NEW GRAMMAR] <Grammar brief description>"
+labels: new data
+assignees: rustam-azimov, vdshk
```
Comment (vadyushkins) [inline .github/ISSUE_TEMPLATE/grammar-add-template.md:6]:
```suggestion
assignees: rustam-azimov, vadyushkins
```

File: .github/ISSUE_TEMPLATE/graph-add-template.md [hunk 6]
```diff
@@ -2,8 +2,8 @@
 name: ADD NEW GRAPH
 about: Suggest new graph to this project
 title: "[NEW GRAPH] <Graph brief description>"
-labels: new graph
-assignees: vdshk
+labels: new data
+assignees: rustam-azimov, vdshk
```
Comment (vadyushkins) [inline .github/ISSUE_TEMPLATE/graph-add-template.md:6]:
```suggestion
assignees: rustam-azimov, vadyushkins
```

**Review by vadyushkins — COMMENTED** [review]
Correct typo

File: cfpq_data/graphs/utils/edges_statistics.py (hunk 2 above)
Comment (vadyushkins) [reply cfpq_data/graphs/utils/edges_statistics.py]: > Yes, I agree for `nested_parentheses_grammar_from_graph` and `dyck_grammar_from_graph`. However, `java_points_to_grammar_from_graph` automatically finds set of fields used in java programs corresponding to our java graphs. For now, only web description of java graphs have information about which fields they have. You think `java_points_to_grammar_from_graph` should be also removed?

Oh, now I see, the `java_points_to_grammar_from_graph` function is unambiguous. I think we should keep it.

**Review by vadyushkins — APPROVED** [review]

**Pull request merged**